projects/saturn/src/main/java/io/fairspace/saturn/services/views/ViewStoreReader.java

```diff
@@ -388,13 +388,20 @@ List<Map<String, Set<ValueDTO>>> retrieveJoinTableRows(
                 joinView.include.stream()
                         .filter(column -> !valueSetProperties.contains(column)))
                 .collect(Collectors.toList());
+
+        // Limit related items by 50, it is only for visualisation and becomes problematic
+        // with thousands of related items.
+        // It will have impact on the column filters, items outside of the limit can't be found.
+        // In these cases the user should make a more narrow selection using facets before using
+        // column filters.
         var query = connection.prepareStatement(
                 "select " + projectionColumns.stream().map(String::toLowerCase).collect(Collectors.joining(", ")) +
                         " from " + joinedTable.name + " j " +
                         " where exists (" +
                         "   select * from " + joinTable.name + " jt " +
                         "   where jt." + idColumn(joinView.view).name + " = j.id" +
-                        "   and jt." + idColumn(view).name + " = ? )"
+                        "   and jt." + idColumn(view).name + " = ? )" +
+                        "limit 50"
```

Comment:
Could we make it a variable to make it easier to spot?

Reply:
you mean replace '50' with a more meaningful name?

Reply:
yeah, just like a constant so it is clear that this is the current setup